src/Microsoft.Identity.Web/CertificateManagement/DefaultCertificateLoader.cs

```diff
@@ -0,0 +1,163 @@
+            using (X509Store x509Store = new X509Store(
+                 certificateStoreName,
+                 certificateStoreLocation))
+            {
+                var by = X509FindType.FindBySubjectDistinguishedName;
+                cert = FindCertificateByCriterium(x509Store, by, certificateSubjectDistinguishedName);
+            }
+
+            return cert;
+        }
+
+        private static X509Certificate2 LoadFromPath(
+            string certificateFileName,
+            string password = null)
+        {
+            return new X509Certificate2(
+                certificateFileName,
+                password,
+                X509KeyStorageFlags.EphemeralKeySet);
+        }
+
+        private static void ParseStoreLocationAndName(string storeDescription, ref StoreLocation certificateStoreLocation, ref StoreName certificateStoreName)
+        {
+            string[] path = storeDescription.Split('/');
+
+            if (path.Length == 2)
+            {
+                if (path.Length != 2
+                    || !Enum.TryParse<StoreLocation>(path[0], true, out certificateStoreLocation)
+                    || !Enum.TryParse<StoreName>(path[1], true, out certificateStoreName))
+                {
+                    throw new ArgumentException("store should be of the form 'StoreLocation/StoreName' with StoreLocation begin 'CurrentUser' or 'CurrentMachine'"
+                        + $" and StoreName begin '' or in '{string.Join(", ", typeof(StoreName).GetEnumNames())}'");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Find a certificate by criteria.
+        /// </summary>
+        /// <param name="x509Store"></param>
+        /// <param name="identifierCriterium"></param>
+        /// <param name="certificateIdentifier"></param>
+        /// <returns></returns>
+        private static X509Certificate2 FindCertificateByCriterium(
+            X509Store x509Store,
+            X509FindType identifierCriterium,
+            string certificateIdentifier)
+        {
+            x509Store.Open(OpenFlags.ReadOnly);
+
+            X509Certificate2Collection certCollection = x509Store.Certificates;
+
+            // Find unexpired certificates.
+            X509Certificate2Collection currentCerts = certCollection.Find(X509FindType.FindByTimeValid, DateTime.Now, false);
+
+            // From the collection of unexpired certificates, find the ones with the correct name.
+            X509Certificate2Collection signingCert = currentCerts.Find(identifierCriterium, certificateIdentifier, false);
+
+            // Return the first certificate in the collection, has the right name and is current.
+            var cert = signingCert.OfType<X509Certificate2>().OrderByDescending(c => c.NotBefore).FirstOrDefault();
+            return cert;
+        }
+
+        internal /*for test only*/ static X509Certificate2 LoadFirstCertificate(IEnumerable<CertificateDescription> certificateDescription)
```

Comment:
I wonder if we shouldn't also have a method LoadFirstCertificate (non static) on ICertificateLoader interface (and make it public). Let's not do it for the moment, but only if customers ask to implement their own ICertificateLoader (not planned for the moment)